primary/src/block_synchronizer/handler.rs

```diff
@@ -38,14 +38,18 @@ pub enum Error {
 
     #[error("Timed out while waiting for {block_id} to become available after submitting for processing")]
     BlockDeliveryTimeout { block_id: CertificateDigest },
+
+    #[error("Payload for block with {block_id} couldn't be synchronized")]
+    PayloadSyncError { block_id: CertificateDigest },
```

Comment:
We could be a lot more precise about diagnosing what went wrong here simply by embedding the inner `SyncError` in the `PayloadSyncError` constructor.